scrape/manager_test.go

```diff
@@ -714,3 +720,94 @@ scrape_configs:
 	reload(scrapeManager, cfg2)
 	require.ElementsMatch(t, []string{"job1", "job3"}, scrapeManager.ScrapePools())
 }
+
+// The test can be improved
+func TestXXX(t *testing.T) {
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+
+	jobName := "my-job"
+	fileTargetURL := "foo:9876"
+
+	// Set up the discovery manager
+	discoveryManager := discovery.NewManager(ctx, log.NewNopLogger())
+	// TODO: 5s by default (maybe we can also reduce scrape manager's DiscoveryReloadInterval)
+	// discoveryManager.updatert = 100 * time.Millisecond
+	go discoveryManager.Run()
+
+	// Set up the scrape manager
+	testRegistry := prometheus.NewRegistry()
+	scrapeManager, err := NewManager(
+		&Options{DiscoveryReloadInterval: model.Duration(100 * time.Millisecond)},
+		nil,
+		nopAppendable{},
+		testRegistry,
+	)
+	require.NoError(t, err)
+	go scrapeManager.Run(discoveryManager.SyncCh())
+	defer scrapeManager.Stop()
+
+	targets := fmt.Sprintf(`
+[{
+	"targets": [
+		"%s"
+	]
+}]
+`, fileTargetURL)
+
+	targetsFile, err := os.CreateTemp("", "*targets.json")
+	require.NoError(t, err)
+	_, err = targetsFile.WriteString(targets)
+	require.NoError(t, err)
+
+	baseCfg := `
+scrape_configs:
+- job_name: %s
+  file_sd_configs:
+  - files:
+    - %s
+`
+	setSDFile := func(sDFile string) {
+		cfg := loadConfiguration(t, fmt.Sprintf(baseCfg, jobName, sDFile))
+		require.NoError(t, err)
+
+		require.NoError(t, scrapeManager.ApplyConfig(cfg))
+
+		c := make(map[string]discovery.Configs)
+		scfgs, err := cfg.GetScrapeConfigs()
+		require.NoError(t, err)
+		for _, v := range scfgs {
+			c[v.JobName] = v.ServiceDiscoveryConfigs
+		}
+		require.NoError(t, discoveryManager.ApplyConfig(c))
+	}
+
+	// Apply the initial config with an existing file
+	setSDFile(targetsFile.Name())
+
+	// Make sure the job's targets are taken into account
+	require.EventuallyWithT(t, func(c *assert.CollectT) {
```

Comment:
I have another version for this not using 'assert' but it isn't better.

maybe there is already a waitfor util, that I'm not aware of, for tests